lore/io/__init__.py

```diff
@@ -138,11 +139,37 @@ def download(remote_url, local_path=None, cache=True, extract=False):
 
     else:
         local_path = temp_path
-        
     return local_path
 
 
-def upload(local_path, remote_path=None):
+# Note: This can be rewritten in a more efficient way
+# https://stackoverflow.com/questions/11426560/amazon-s3-boto-how-to-delete-folder
+def delete_folder(remote_url):
+    if remote_url is None:
+        raise ValueError("remote_url cannot be None")
+    else:
+        remote_url = prefix_remote_root(remote_url)
+        keys = bucket.objects.filter(Prefix=remote_url + '/')
+        empty = True
+
+        for key in keys:
+            empty = False
+            key.delete()
+
+        if empty:
+            logger.info('Remote was not a folder')
+
+
+def upload_object(obj, remote_path=None):
+    if remote_path is None:
+        raise ValueError("remote_path cannot be None when uploading objects")
+    else:
+        with tempfile.NamedTemporaryFile() as f:
+            pickle.dump(obj, f)
```

Comment:
Is it possible to avoid the tempfile with `pickle.dumps` and uploading the result directly to s3? I'm curious about the memory requirements of `dump` vs `dumps` if it matters at all.

Reply:
Definitely is. Just wanted to avoid premature optimization for now.